### ISSUE TITLE
Copy nullable state through implicit and explicit conversions

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#32599]
The following should report warnings for `u1.Item2` and `u2.Item2` only:
```C#
class Program
{
    static void F1((string?, string) t1)
    {
        (object?, object?) u1 = t1;
        u1.Item1.ToString(); // warn
        u1.Item2.ToString();
    }

    static void F2((object?, object) t2)
    {
        var u2 = ((string?, string?))t2;
        u2.Item1.ToString(); // warn
        u2.Item2.ToString();
    }
}
```

**Comment by cston** [comment]
Another example, from discussion with @jcouv:
```C#
class A
{
    public static implicit operator B?(A a) => null;
}

class B
{
}

class Program
{
    static void F(A a, B b)
    {
        (B, B) t = (a, b);
        t.Item1.ToString(); // warning: maybe null
        t.Item2.ToString();
    }
}
```

**Comment by BillWagner** [comment]
@cston Is there a typo in the description above? Should the warning be issued only on `Item1` instead of `Item2`?

**Comment by cston** [comment]
@BillWagner,

> Should the warning be issued only on Item1 instead of Item2

Yes, I think you are right. I've updated the description, thanks.
